presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceStreamReader.java

```diff
@@ -105,18 +106,30 @@ public String toString()
                 .toString();
     }
 
-    public static int computeTruncatedLength(Slice slice, int offset, int length, Type type)
+    public static int getMaxCodePointCount(Type type)
     {
-        // calculate truncated length
-        int truncatedLength = length;
         if (isVarcharType(type)) {
             VarcharType varcharType = (VarcharType) type;
-            int codePointCount = varcharType.isUnbounded() ? length : varcharType.getLengthSafe();
-            truncatedLength = byteCount(slice, offset, length, codePointCount);
+            return varcharType.isUnbounded() ? -1 : varcharType.getLengthSafe();
+        }
+        if (isCharType(type)) {
+            return ((CharType) type).getLength();
         }
-        else if (isCharType(type)) {
+        if (isVarbinaryType(type)) {
+            return -1;
+        }
+        throw new IllegalArgumentException("Unsupported encoding " + type.getDisplayName());
+    }
+
+    public static int computeTruncatedLength(Slice slice, int offset, int length, int maxCodePointCount, boolean isCharType)
+    {
+        int truncatedLength = length;
+        if (isCharType) {
             // truncate the characters and then remove the trailing white spaces
-            truncatedLength = byteCountWithoutTrailingSpace(slice, offset, length, ((CharType) type).getLength());
+            truncatedLength = byteCountWithoutTrailingSpace(slice, offset, length, maxCodePointCount);
```

Comment:
You can just return directly from here... which would make the else redundant (and return directly from the next block).